### PR TITLE
Add upsert support for pgvector

### LINKED ISSUE #6866: PGVector is Lacking Basic Features
### Feature request

PGVector lacks Upsert and deletion capabilities.

I have to custom create this functionality.

### Motivation

I want to use PGVector because it's easy to implement and I don't require to deal with DB vector providers.

### Your contribution

If you deem this useful I will try propose a pull request.

### PR DESCRIPTION
**Pull request by lokeshdevnani**
## Description
Currently, there is no support for updates or upserts to PGVector through langchain.
This PR adds that support by leveraging `on_conflict_do_upsert`

This fixes https://github.com/langchain-ai/langchain/issues/6866

Maintainers: @rlancemartin, @eyurtsev

Twitter handle: [@lokeshdevnani](https://twitter.com/lokeshdevnani)

### CODE REVIEW

**Comment by eyurtsev** [comment]
@rlancemartin did we make a decision on whether `upsert` gets its own methods?

**Comment by leo-gan** [comment]
@eyurtsev  Could you, please, review it?

File: libs/langchain/langchain/vectorstores/pgvector.py [hunk 1]
```diff
@@ -253,6 +282,7 @@ def add_texts(
         texts: Iterable[str],
         metadatas: Optional[List[dict]] = None,
         ids: Optional[List[str]] = None,
+        upsert: bool = False,
         **kwargs: Any,
     ) -> List[str]:
         """Run more texts through the embeddings and add to the vectorstore.
```
Comment (efriis) [inline libs/langchain/langchain/vectorstores/pgvector.py:288]: can you update docstrings with the new arg?

**Review by efriis — COMMENTED** [review]
2 small changes and fix linting please!

**Comment by baskaryan** [comment]
Apologies for the slow review! Pr has a number of merge conflicts, happy to re-review if you'd like to resolve

**Comment by hwchase17** [comment]
moved to own package

**Pull request closed without merging**